tests/lib/assertions.py

```diff
@@ -14,7 +17,10 @@ def assert_201_response(resp, messageBatchReference):
         assert response.get("attributes").get("messageBatchReference") == messageBatchReference
 
     @staticmethod
-    def assert_error_with_optional_correlation_id(resp, code, error, correlation_id):
+    def assert_error_with_optional_correlation_id(resp, code, error, correlation_id, too_many_requests_tests=False):
+        if (too_many_requests_tests is False and resp.status_code == 429):
```

Comment:
You can simplify this - remove the `too_many_requests` param and instead use the expected code value - so something like `if code != 429 and resp.status_code == 429:`